clinical/src/main/java/com/hartwig/actin/clinical/curation/CurationDatabaseValidator.java

```diff
@@ -0,0 +1,119 @@
+                allValid = false;
+                LOGGER.warn(" Invalid primary tumor doids configured for '{}': {}", primaryTumorConfig.input(), primaryTumorConfig.doids());
+            }
+        }
+        return allValid;
+    }
+
+    @VisibleForTesting
+    static boolean validateSecondPrimaryConfigs(@NotNull List<SecondPrimaryConfig> secondPrimaryConfigs,
+            @NotNull DoidModel doidModel) {
+        boolean allValid = true;
+        for (SecondPrimaryConfig secondPrimaryConfig : secondPrimaryConfigs) {
+            if (!hasValidDoids(secondPrimaryConfig.curated().doids(), doidModel, CANCER_PARENT_DOID)) {
+                allValid = false;
+                LOGGER.warn(" Invalid second primary doids configured for '{}': {}",
+                        secondPrimaryConfig.input(),
+                        secondPrimaryConfig.curated().doids());
+            }
+        }
+        return allValid;
+    }
+
+    @VisibleForTesting
+    static boolean validateNonOncologicalHistoryConfigs(@NotNull List<NonOncologicalHistoryConfig> nonOncologicalHistoryConfigs,
+            @NotNull DoidModel doidModel) {
+        boolean allValid = true;
+        for (NonOncologicalHistoryConfig nonOncologicalHistoryConfig : nonOncologicalHistoryConfigs) {
+            Object curated = nonOncologicalHistoryConfig.curated();
+            if (curated instanceof PriorOtherCondition) {
+                PriorOtherCondition priorOtherCondition = (PriorOtherCondition) curated;
+                if (!hasValidDoids(priorOtherCondition.doids(), doidModel, GENERIC_PARENT_DOID)) {
+                    allValid = false;
+                    LOGGER.warn(" Invalid prior other condition doids configured for '{}': {}",
+                            nonOncologicalHistoryConfig.input(),
+                            priorOtherCondition.doids());
+                }
+            }
+        }
+        return allValid;
+    }
+
+    @VisibleForTesting
+    static boolean validateIntoleranceConfigs(@NotNull List<IntoleranceConfig> intoleranceConfigs, @NotNull DoidModel doidModel) {
+        boolean allValid = true;
+        for (IntoleranceConfig intoleranceConfig : intoleranceConfigs) {
+            if (!hasValidDoids(intoleranceConfig.doids(), doidModel, GENERIC_PARENT_DOID)) {
+                allValid = false;
+                LOGGER.warn(" Invalid intolerance doids configured for '{}': {}", intoleranceConfig.input(), intoleranceConfig.doids());
+            }
+        }
+        return allValid;
+    }
+
+    @VisibleForTesting
+    static boolean hasValidDoids(@NotNull Set<String> doids, @NotNull DoidModel doidModel, @NotNull String expectedParentDoid) {
+        if (doids.isEmpty()) {
+            return false;
+        }
+
+        for (String doid : doids) {
+            if (!doidModel.doidWithParents(doid).contains(expectedParentDoid)) {
+                return false;
+            }
+        }
+        return true;
```

Comment:
Alternatively:
```suggestion
        return doids.stream().allMatch(doid -> doidModel.doidWithParents(doid).contains(expectedParentDoid));
```

Reply:
Though we'd have to write the log message before returning if you take my other suggestion.

Reply:
Have taken over the streaming solution, but not sure about the logging. I don't like passing params into more deep code just for the sake of generating a log message. But do have plans to approach the whole ticket somewhat different so not too relevant.